eclair-core/src/test/scala/fr/acinq/eclair/wire/TlvCodecsSpec.scala

```diff
@@ -308,11 +306,25 @@ class TlvCodecsSpec extends FunSuite {
     assert(lengthPrefixedTestTlvStreamCodec.encode(stream).require.toByteVector === hex"0f 01012a 0b012b 0d012a fd00fe02002a")
   }
 
+  test("encode all even generic tlv types and decode even types above high range") {
+    val lowRangeOdd = TlvStream[TestTlv](records = Nil, unknown = Seq(GenericTlv(123, hex"0a")))
+    val lowRangeEven = TlvStream[TestTlv](records = Nil, unknown = Seq(GenericTlv(124, hex"0a")))
+    val highRangeOdd = TlvStream[TestTlv](records = Nil, unknown = Seq(GenericTlv(67876545677L, hex"0a")))
+    val highRangeEven = TlvStream[TestTlv](records = Nil, unknown = Seq(GenericTlv(67876545678L, hex"0a")))
+
+    assert(testTlvStreamCodec.encode(lowRangeOdd).isSuccessful)
+    assert(testTlvStreamCodec.encode(lowRangeEven).isSuccessful)
+    assert(testTlvStreamCodec.encode(highRangeOdd).isSuccessful)
+    assert(testTlvStreamCodec.encode(highRangeEven).isSuccessful)
+
+    assert(testTlvStreamCodec.decode(testTlvStreamCodec.encode(lowRangeOdd).require).isSuccessful)
+    assert(testTlvStreamCodec.decode(testTlvStreamCodec.encode(lowRangeEven).require).isFailure)
+    assert(testTlvStreamCodec.decode(testTlvStreamCodec.encode(highRangeOdd).require).isSuccessful)
+    assert(testTlvStreamCodec.decode(testTlvStreamCodec.encode(highRangeEven).require).isSuccessful)
+  }
+
   test("encode invalid tlv stream") {
     val testCases = Seq(
-      // Unknown even type.
```

Comment:
I think you can restore those (once we reject encoding tlv types that are not in the > 2^16 range - see my other comment)